Sources/Rendering/Core/ColorTransferFunction/test/testColorTransferFunction.js

```diff
@@ -118,3 +120,41 @@ test.onlyIfWebGL('Test Interpolate Scalars Before Colors', (t) => {
   });
   renderWindow.render();
 });
+
+test('Test discretized color transfer function', (t) => {
+  const ctf = vtkColorTransferFunction.newInstance({
+    discretize: true,
+    numberOfValues: 5,
+  });
+
+  ctf.addRGBPoint(0.0, 1.0, 0.0, 0.0);
+  ctf.addRGBPoint(1.0, 0.0, 1.0, 0.0);
+  ctf.addRGBPoint(2.0, 0.0, 0.0, 1.0);
+  ctf.setMappingRange(0.0, 1.0);
+
+  const testValues = [-0.1, 0.0, 0.1, 0.19, 0.2, 0.59, 0.6, 0.9, 1.0, 1.1];
+  const expectedRGB = [
+    [1, 0, 0],
+    [1, 0, 0],
+    [1, 0, 0],
+    [1, 0, 0],
+    [0.5, 0.5, 0],
+    [0, 1, 0],
+    [0, 0.5, 0.5],
+    [0, 0, 1],
+    [0, 0, 1],
+    [0, 0, 1],
+  ];
```

Comment:
does this test fail without your fix ? for what index ? 0 or 1 ?

Reply:
It fails for 2 and 3.